subsys/bluetooth/audio/ascs.c

```diff
@@ -558,55 +568,41 @@ static int ascs_ep_get_status(struct bt_bap_ep *ep, struct net_buf_simple *buf)
 	return 0;
 }
 
-static int ascs_iso_accept(const struct bt_iso_accept_info *info,
-				      struct bt_iso_chan **iso_chan)
+static int ascs_iso_accept(const struct bt_iso_accept_info *info, struct bt_iso_chan **iso_chan)
 {
-	LOG_DBG("acl %p", (void *)info->acl);
-
-	for (size_t i = 0U; i < ARRAY_SIZE(enabling); i++) {
-		struct bt_bap_stream *c = enabling[i];
+	struct bt_ascs *ascs = ascs_get(info->acl);
+	struct bt_ascs_ase *ase;
 
-		if (c != NULL && c->ep->cig_id == info->cig_id && c->ep->cis_id == info->cis_id) {
-			*iso_chan = &enabling[i]->ep->iso->chan;
-			enabling[i] = NULL;
+	LOG_DBG("ascs %p", ascs);
 
-			LOG_DBG("iso_chan %p", *iso_chan);
+	SYS_SLIST_FOR_EACH_CONTAINER(&ascs->ases, ase, node) {
+		enum bt_bap_ep_state state;
+		struct bt_iso_chan *chan;
 
-			return 0;
+		if (ase->ep.cig_id != info->cig_id || ase->ep.cis_id != info->cis_id) {
+			continue;
 		}
-	}
-
-	LOG_ERR("No channel listening");
-
-	return -EPERM;
-}
-
-static int ascs_iso_listen(struct bt_bap_stream *stream)
-{
-	struct bt_bap_stream **free_stream = NULL;
-
 
-	LOG_DBG("stream %p conn %p", stream, (void *)stream->conn);
-
-	for (size_t i = 0U; i < ARRAY_SIZE(enabling); i++) {
-		if (enabling[i] == stream) {
-			return 0;
+		state = ascs_ep_get_state(&ase->ep);
+		if (state != BT_BAP_EP_STATE_ENABLING && state != BT_BAP_EP_STATE_QOS_CONFIGURED) {
+			LOG_WRN("ase %p cannot accept ISO connection", ase);
+			break;
 		}
 
-		if (enabling[i] == NULL && free_stream == NULL) {
-			free_stream = &enabling[i];
+		chan = &ase->ep.iso->chan;
+		if (chan->iso != NULL) {
+			LOG_WRN("ase %p chan %p already connected", ase, chan);
+			break;
 		}
-	}
 
-	if (free_stream != NULL) {
-		*free_stream = stream;
+		*iso_chan = chan;
+
+		LOG_DBG("iso_chan %p", *iso_chan);
 
 		return 0;
 	}
 
-	LOG_ERR("Unable to listen: no slot left");
-
-	return -ENOSPC;
+	return -EPERM;
```

Comment:
Hmm, in libc `EPERM` is defined as
```c
#define EPERM 1         /**< Not owner */
```

Perhaps `EACCES` is better suited here?
```c
#define EACCES 13       /**< Permission denied */
```

Reply:
The return code was already there, did not changed it. But I agree, the `EACCES` would be better.